src/steps/05-PerformanceRequirements/DOW/DOWLandingPage.vue

```diff
@@ -92,6 +116,123 @@ export default class DOWLandingPage extends Vue {
+   * 2. If the user needs an architectural solution AND
+   *      provides a detailed statement.
+   * 3. If the user needs an architectural solution AND
+   *      identifies at least one classification level that needs a solution.
+   */
+  async setArchitecturalDesignProperties(): Promise<void> {
+    const currEnv = CurrentEnvironment.currentEnvironment;
+    const currEnvArchNeeds = await CurrentEnvironment.getCurrentEnvironmentArchitecturalNeeds();
+    // is complete check
+    const reqSectionIndex =this.requirementSections
+      .findIndex(reqSection => reqSection.section === "ArchitecturalDesign")
+    this.requirementSections[reqSectionIndex].isComplete =
+        (currEnv.needs_architectural_design_services === "NO") ||
+        ((currEnvArchNeeds.statement?.trim().length > 0) &&
+            (currEnvArchNeeds.data_classification_levels?.length > 0))
+    // label
+    let label = "";
+    if(currEnvArchNeeds.data_classification_levels?.length > 0 ) {
+      const classificationLevels = await classificationRequirements
+        .getAllClassificationLevels();
+      label = "Required for ";
+      const selectedLevels = currEnvArchNeeds.data_classification_levels as unknown as string[];
+      selectedLevels.forEach((selectedLabel, index) => {
+        const selectedClasslevelObj = classificationLevels
+          .find(classificationLevel =>
+            classificationLevel.sys_id === selectedLabel) as ClassificationLevelDTO;
+        label = label + selectedClasslevelObj.display
+        if ((selectedLevels.length > 1) && (index === selectedLevels.length - 2)) {
+          label = label + " and ";
+        } else if (index < selectedLevels.length - 1) {
+          label = label + ", ";
+        }
+      })
+      label = label + (selectedLevels.length > 1 ? " environments" : " environment");
+    } else if (currEnv.needs_architectural_design_services === "NO") {
+      label = "No requirements";
+    }
+    if (label !== "") {
+      this.requirementSections[reqSectionIndex].label = label;
+    }
+  }
+
+  /**
+   * Checks all three sections and if the user explicitly selects no requirements across all the
+   * required sections. This check does not apply if the user does not start or completes one or
+   * more of the required sections.
+   */
+  checkDisplayWarning(): void {
+    const currEnv = CurrentEnvironment.currentEnvironment;
+    let allRequiredSectionsComplete = true;
+    ["ReplicateOptimize", "ArchitecturalDesign", "XaaS"].forEach(requiredSection => {
+      const reqSectionIndex =this.requirementSections
+        .findIndex(reqSection => reqSection.section === requiredSection);
+      if(allRequiredSectionsComplete && !this.requirementSections[reqSectionIndex].isComplete ) {
+        allRequiredSectionsComplete = false;
+      }
+    })
+    if(allRequiredSectionsComplete &&
+        currEnv.current_environment_replicated_optimized === "NO" &&
+        currEnv.needs_architectural_design_services === "NO" &&
```

Comment:
For this and the other issue marked here, since this is functionally working, should we continue to block this PR or move forward by creating a refactor task? Your call.